### DISCUSSION TITLE
Jobrunr 5.2.0 in GCP appengine

### DISCUSSION THREAD

**Discussion by iamsuryarej** [jobrunr/jobrunr#720] [Q&A]
Hi I am using jobrunr 5.2.0 in my java 8 appengine project. My aim is to export data from cloud storage bucket to Bigquery. The export is working fine when I am not using jobrunr but  i am getting this error whenever I am using jobrunr
`com.google.apphosting.api.ApiProxy$CallNotFoundException: Can't make API call app_identity_service.GetDefaultGcsBucketName in a thread that is neither the original request thread nor a thread created by ThreadManager
`
also this exception
`Current thread is not associated with any request and is not a background thread`

My java code looks like this

`jobScheduler.enqueue(()->bigQueryExportServicev2.export(entityKindId1));
`

**Comment by rdehuyss** [comment]
I don't think JobRunr will work in GCP as they do some strange stuff related to threading...

If this is a requirement for you, feel free to reach out for a Pro version and I can see what I can do.

**Reply by iamsuryarej** [reply]
I am interested but my requirement is that my code should cloud cloud agnostic. We are building a multicloud web app.

**Reply by rdehuyss** [reply]
Well, it works for all the other clouds, GCP is the only cloud having restrictions regarding thread management.

**Reply by iamsuryarej** [reply]
Yup thats what i experienced . Is there any solution to this.

**Comment by rdehuyss** [comment]
> feel free to reach out for a Pro version and I can see what I can do.

You can get into touch with me via the website 😄
